docs/cvl/statements.md

```diff
@@ -93,6 +94,38 @@ Unlike Solidity's `assert` and `require`, the CVL syntax for `assert` and
 `require` does not require parentheses around the expression and message.
 ```
 
+(satisfy)=
+`satisfy` statements
+--------------------
+
+A `satisfy` statement is used to check that the rule can be executed in such a
+way that the `satisfy` statement is true.  A rule with a `satisfy` statement is
+describing a scenario and may not contain `assert` statements.  We require that
+each rule ends with a `satisfy` statement or an `assert` statement.
+
+For each `satisfy` statement, the Certora verifier will produce a witness for a
+valid execution of the rule.  It will show an execution trace containing values
+for each input variable and each state variable where all `require` and `satisfy`
+statements are executed successfully.  In case there is no such execution, for
+example if the `require` statements are already inconsistent or if a solidity
+function always reverts, an error is reported.
+
+If the rule contains multiple `satisfy` statements, then all executed `satisfy`
+statements must hold.   However, a `satisfy` statement on a conditional branch that
+is not executed, does not need to hold.  The verifier produces multiple witnesses
+such that for every `satisfy` statement there is a witness that executes this
+statement.
+
+If for at least one `satisfy` statement there is no execution path on which it
+holds, an error is reported.  If all `satisfy` statements can be fulfilled on
+at least one path, the rule succeeds.
+
+```{note}
+A success does only guarantee that there is some execution starting in some
+arbitrary state.  It is not possible to check that there is an execution for
```

Comment:
Doesn't `assert` check that the expression holds for every starting state? Should we direct the reader to try specifying this with assert instead if this is their goal?

Reply:
OK sounds reasonable.